libpod/container_internal_linux.go

```diff
@@ -2339,13 +2339,39 @@ func (c *Container) createSecretMountDir() error {
 		oldUmask := umask.Set(0)
 		defer umask.Set(oldUmask)
 
-		if err := os.MkdirAll(src, 0644); err != nil {
+		srcMountPermission := 0644
+		// If env is rootless give directory execute permission
+		// Non-root ( uid != 0 user can only access dir if it has `x`
+		if rootless.IsRootless() {
+			srcMountPermission = 0744
```

Comment:
Permissions should just be 755. In rootful and rootless. 644 makes no sense.

Reply:
@rhatdan  I am confused about `755` , I just escalated mount to `644` -> `744` for `non-root` user so that only mount owner could access this dir. I thought thats what author thought initially , if we are going to bump this to `755` all the created users on container would have access to secrets. Not sure but could this be a potential security flaw since they are mounted secrets ? WDYT ?

But I can change this to  `755` if that is expected behavior.

Reply:
@rhatdan I verified this with `docker swarm` it has analogous behavior, in places even tighter permissions. For instance when owner is root , permissions is `444` and adds execute when non-root user is specified or in some use-cases people are doing it manually. https://stackoverflow.com/questions/53385930/docker-secrets-with-non-root-user

I have added a test for rootless user and it seems to be passing also I have verified this manually just pasting output here
```bash
SECRET="mysecret"
echo a_secret | podman secret create "$SECRET" -
$ podman run --secret "$SECRET" --rm --user 200:200 -i -t docker.io/library/debian sh
$ cat /run/secrets/mysecret
a_secret
$ 
```

Reply:
@rhatdan Let me know if this needs to be closed as discussed here https://github.com/containers/common/issues/555

Reply:
My point is the directory where the secret is stored and all of the paths up to it should be (root:root) 0755 so that all users within the container can search through the directories to get to the secret file.  The file with the secrets should be 0444 so all users of the container can read it.

I guess you could argue that the secret `file` should be owned by the default user and be 0600, but I am fine with making it readable by all users.

I see the secret file being something that can be used by processes within the container, but not something that should be saved to the image if I was building a container image.

Reply:
@rhatdan Ah my bad i misunderstood few things. I agree `0755` for `secrets` dir makes sense. However actual  file underneath should be `0444`. I have done the required changes for mounted `run/secrets` dir.

But as discussed containers/common#555 this PR might overlap a bit with what @ashley-cui is already doing. Not sure what needs to be done.  But this PR would definitely fix  issue being faced in containers/common#555

Reply:
If you don't mind, I'd be in favor of closing this PR and using the one I have coming up, since I'll have to move all these changes anyway

Reply:
No problem.  @flouthoc Thanks for the effort.  Please work with @ashley-cui and her PR since she is in charge of this issue.  And make sure it matches your expectations.

Reply:
@flouthoc Thanks for giving this a shot though!! I'll @ you when I open my PR do you can take a look :)

Reply:
@ashley-cui @rhatdan  Sure no issues. 😄